airgen/src/lib.rs

```diff
@@ -57,42 +57,45 @@ pub fn compile(input: AnalysisASMFile) -> PILGraph {
     };
 
     // get a list of all machines to instantiate. The order does not matter.
-    let mut queue = vec![(main_location.clone(), main_ty.clone())];
+    let mut queue = vec![(main_location.clone(), main_ty.clone(), vec![])];
 
-    let mut instances = vec![];
+    // map instance location to (type, arguments)
+    let mut instances = BTreeMap::default();
 
-    while let Some((location, ty)) = queue.pop() {
+    while let Some((location, ty, param_values)) = queue.pop() {
         let machine = input.items.get(&ty).unwrap().try_to_machine().unwrap();
 
         queue.extend(machine.submachines.iter().map(|def| {
             (
                 // get the absolute name for this submachine
                 location.clone().join(def.name.clone()),
-                // get its type
+                // type
```

Comment:
I think we should clarify here that this is the machine type - since we now also have types for the parameters.